gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/CliMRJobLauncher.java

```diff
@@ -52,15 +53,11 @@ public CliMRJobLauncher(Properties sysConfig, Properties jobConfig)
   @Override
   public int run(String[] args)
       throws Exception {
-    final Properties jobProps = new Properties();
-    // First load system configuration properties
-    jobProps.putAll(this.sysConfig);
-    // Then load job configuration properties that might overwrite system configuration properties
-    jobProps.putAll(this.jobConfig);
+    final Properties jobProps = JobConfigurationUtils.combineSysAndJobProperties(this.sysConfig, this.jobConfig);
 
     Closer closer = Closer.create();
     try {
-      closer.register(new MRJobLauncher(this.sysConfig, jobProps, getConf())).launchJob(null);
```

Comment:
Its not really relevant to this PR, but any reason why this line doesn't use `JobLauncherFactory`